src/main/java/willitconnect/service/EntryChecker.java

```diff
@@ -39,37 +43,66 @@ public EntryChecker(RestTemplate restTemplate) {
         restTemplate.setErrorHandler(new CustomResponseErrorHandler());
     }
 
-    public CheckedEntry check(CheckedEntry e, String proxy, int proxyPort,
-                              String proxyType) {
+    public CheckedEntry check(CheckedEntry e) {
         log.info("checking " + e.getEntry());
+
         if (e.isValidUrl()) {
             try {
+                ClientHttpRequestFactory oldFactory = null;
+                if ( null != e.getHttpProxy() ) {
```

Comment:
Is there a better way to do this? It's hard to test (read as: is not really tested) and super invasive. I couldn't think of a way to inject it without swapping.